includes/class-freemius.php

```diff
@@ -381,7 +381,7 @@ private function __construct( $module_id, $slug = false, $is_init = false ) {
             $this->_plugin_main_file_path = $this->_find_caller_plugin_file( $is_init );
             $this->_plugin_dir_path       = plugin_dir_path( $this->_plugin_main_file_path );
             $this->_plugin_basename       = $this->get_plugin_basename();
-            $this->_free_plugin_basename  = str_replace( '-premium/', '/', $this->_plugin_basename );
+            $this->_free_plugin_basename  = ( $this->_slug . '/' . basename( $this->_plugin_basename ) );

```

Comment:
@fajardoleo I think that this change is a little risky and things may work unexpectedly if the admin changes the free version's folder name. We may want to update it in `parse_settings()` after loading the info if we are running the free or paid product, and doing something like that:
```
if ( ! $this->is_premium() ) {
    $this->_free_plugin_basename = $this->_plugin_basename;
}
```

But we need to verify that the `_free_plugin_basename` isn't actually used in the logic between the constructor and the parsing of the settings.